lib/Parse/ParseExpr.cpp

```diff
@@ -3577,6 +3580,42 @@ Parser::DeferredParseBoundsExpression(std::unique_ptr<CachedTokens> Toks,
   return Error;
 }
 
+// Callback for parsing the return bounds expression in Toks.
+bool Parser::ParseBoundsCallback(void *P,
+                                 std::unique_ptr<CachedTokens> Toks,
+                                 ArrayRef<ParmVarDecl *> Params,
+                                 BoundsAnnotations &Result,
+                                 const Declarator &D) {
+  assert(P);
+  Parser *TheParser = (Parser *) P;
+
+  // Set up function prototype scope again and put parameters back in.
+  unsigned PrototypeScopeFlag =
+      Scope::FunctionPrototypeScope | Scope::DeclScope |
+      (D.isFunctionDeclaratorAFunctionDeclaration()
+            ? Scope::FunctionDeclarationScope
+            : 0);
+
+  PrototypeScopeFlag |=
+      (D.getDeclSpec().isCheckedSpecified()
+            ? Scope::CheckedScope
+            : (D.getDeclSpec().isUncheckedSpecified() ? Scope::UncheckedScope
+                                                      : 0));
+
+  ParseScope PrototypeScope(TheParser, PrototypeScopeFlag);
+  TheParser->Actions.ActOnSetupParametersAgain(TheParser->Actions.CurScope, Params);
+  bool Err = TheParser->DeferredParseBoundsExpression(std::move(Toks), Result, D);
+  PrototypeScope.Exit();
+  return Err;
+}
+
+ExprResult Parser::ParseReturnValueExpression() {
+  assert(Tok.is(tok::kw__Return_value) &&
+         "Not bounds  value expression");
```

Comment:
Double space between bounds and value in the message "Not bounds value expression"